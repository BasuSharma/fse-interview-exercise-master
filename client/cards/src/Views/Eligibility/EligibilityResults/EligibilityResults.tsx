import React from "react";
import styled from "styled-components";
import View from "../../../DesignSystem/View";
import Card from "../../../DesignSystem/Card";

const ResultsWrapper = styled.div`
  flex: 1 1 auto;
  padding-top: 48px;
  justify-content: center;
  margin: 0 -8px;
  display: flex;
  flex-wrap: wrap;
`;

const HeadingText = styled.div`
  flex: 1 1 auto;
  width: 80%;
  padding-top: 10px;
  padding-bottom:10px;
  justify-content: center;
  display: flex;
  flex-wrap: wrap;
  border-radius: 1px;
  border: solid .5px #eee;
  min-height: 50px;
  font-size: 25px;
  font-weight: bold;
  text-align: center;
  color: rgb(42, 98, 231, 0.75);
`;

//Component to show results of eligibility check of applicant 
const EligibilityResults = (props: { userName: string, cards: any }) => {
   
    if(props.cards.eligibleCards.length >0)
    {
     return <ResultsWrapper>
        <View>
        <HeadingText> Hi {props.userName}, <br></br> Below are your eligible card/cards </HeadingText>
        {props.cards.eligibleCards.map((card: string) => {
          return <Card key={card}>  {card} </Card>
        })}
      </View>
      <br></br>
      <div><a href="http://localhost:3000/">back</a></div>
      </ResultsWrapper>;
    }
    else{
      return <ResultsWrapper>
        <View>
          <HeadingText> Hi {props.userName}, 
          <br></br>Sorry we don't have any eligible card for you.
          </HeadingText>
      </View>
      <div><a href="http://localhost:3000/">back</a></div>
      </ResultsWrapper>;
    }

 
};
  
export default EligibilityResults;


