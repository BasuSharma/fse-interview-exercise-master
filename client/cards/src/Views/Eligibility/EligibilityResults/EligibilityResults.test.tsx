import React from 'react';
import { screen } from '@testing-library/react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson, { Json } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import EligibilityResults from './EligibilityResults';

Enzyme.configure({ adapter: new Adapter() })

let cardsData = {cards:{
                            eligibleCards:[
                                            "C1",
                                            "C2"
                                        ]
                                        },
                    };


it("renders correctly", () => {
  const testresultcomponent = shallow(<EligibilityResults  userName={"Anjela"} cards= {cardsData.cards} />);
  expect(toJson(testresultcomponent)).toMatchSnapshot();
});
