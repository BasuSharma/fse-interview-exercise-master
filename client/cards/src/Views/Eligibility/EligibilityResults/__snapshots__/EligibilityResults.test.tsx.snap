// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
       Hi 
      Anjela
      , 
      <br />
       Below are your eligible card/cards 
    </styled.div>
    <styled.div
      key="C1"
    >
        
      C1
       
    </styled.div>
    <styled.div
      key="C2"
    >
        
      C2
       
    </styled.div>
  </styled.div>
  <br />
  <div>
    <a
      href="http://localhost:3000/"
    >
      back
    </a>
  </div>
</styled.div>
`;
